libpromises/rlist.c

```diff
@@ -1206,35 +1204,23 @@ void RlistWrite(Writer *writer, const Rlist *list)
     WriterWriteChar(writer, '}');
 }
 
-/* Note: only single quotes are escaped, as they are used in RlistWrite to
-   delimit strings. If double quotes would be escaped, they would be mangled by
-   RlistParseShown */
-
-static void ScalarWrite(Writer *w, const char *s)
+void ScalarWrite(Writer *writer, const char *s, bool quote)
 {
+    if(quote)
+        WriterWriteChar(writer, '"');
```

Comment:
Code style:
* Should be space before expression after if statement.
* No if blocks without braces.

I fixed these for now, but please remember for the future.